pkg/crypto/rings/client.go

```diff
@@ -334,6 +341,18 @@ func GetRingAddressesAtBlock(
 	return activeDelegationsAtHeight
 }
 
+// GetRingFromPubKeys returns a ring constructed from the public keys provided.
+func GetRingFromPubKeys(ringPubKeys []cryptotypes.PubKey) (*ring.Ring, error) {
```

Comment:
Should this be factored out to the `crypto/ring` pkg?

Reply:
Where do you suggest we should put it? FYI it is used in `shannon-sdk` to have a single ring construction codebase.

Reply:
What about pkg/crypto/rings/ring.go?